broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/LogStoragePartitionStep.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.partitions.impl.steps;
+
+import static io.camunda.zeebe.util.Either.left;
+import static io.camunda.zeebe.util.Either.right;
+
+import io.atomix.raft.partition.impl.RaftPartitionServer;
+import io.atomix.raft.zeebe.ZeebeLogAppender;
+import io.camunda.zeebe.broker.system.partitions.PartitionStartupAndTransitionContextImpl;
+import io.camunda.zeebe.broker.system.partitions.PartitionStep;
+import io.camunda.zeebe.logstreams.storage.atomix.AtomixLogStorage;
+import io.camunda.zeebe.util.Either;
+import io.camunda.zeebe.util.sched.future.ActorFuture;
+import io.camunda.zeebe.util.sched.future.CompletableActorFuture;
+
+public class LogStoragePartitionStep implements PartitionStep {
```

Comment:
I assume we extracted it because we want to open that once at some point and only recreate the wrapper logstream?

Reply:
Yes. We might be able to open it only once. Although it is not possible now, because we have to have a read-only storage for followers. But I guess we can find a way to open it once.